lib/skunk/generators/json/simple.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+require "rubycritic/generators/json/simple"
+
+module Skunk
+  module Generator
+    module Json
+      # Generates a JSON report for the analysed modules.
+      class Simple < RubyCritic::Generator::Json::Simple
+        def data
+          {
+            analysed_modules_count: analysed_modules_count,
+            skunk_score_average: skunk_score_average,
+            skunk_score_total: skunk_score_total,
+            worst_pathname: worst&.pathname,
+            worst_score: worst&.skunk_score,
+            files: files
+          }
+        end
+
+        private
+
+        def analysed_modules_count
+          @analysed_modules_count ||= non_test_modules.count
+        end
+
+        def skunk_score_average
+          return 0 if analysed_modules_count.zero?
+
+          (skunk_score_total.to_d / analysed_modules_count).to_f.round(2)
+        end
+
+        def skunk_score_total
+          @skunk_score_total ||= non_test_modules.sum(&:skunk_score)
+        end
+
+        def non_test_modules
+          @non_test_modules ||= @analysed_modules.reject do |a_module|
+            module_path = a_module.pathname.dirname.to_s
+            module_path.start_with?("test", "spec") || module_path.end_with?("test", "spec")
+          end
+        end
+
+        def worst
+          @worst ||= sorted_modules.first
+        end
+
+        def sorted_modules
+          @sorted_modules ||= non_test_modules.sort_by(&:skunk_score).reverse!
+        end
+
+        def files
+          @files ||= sorted_modules.map { |file| row(file) }
+        end
+
+        # :reek:UtilityFunction
+        def row(file)
+          {
+            pathname: file.pathname,
+            skunk_score: file.skunk_score,
+            churn_times_cost: file.churn_times_cost,
+            churn: file.churn,
+            cost: file.cost.round(2),
+            coverage: file.coverage.round(2)
+          }
+        end
```

Comment:
All the private methods are the same in the `lib/skunk/commands/status_reporter.rb` file, but that file is currently tied to generate the `console` report, and to not break anything for now, we are duplicating them